.github/workflows/operatorBuildAndDeploy.yml

```diff
@@ -38,35 +38,143 @@ jobs:
         run: |
           export PATH=$GOROOT/bin:$GOPATH/bin:$PATH
           mage operator:testGenerateClient
-      - name: Build docker - standard and ubi images
+      - name: Unit Tests
+        run: |
+          export PATH=$GOROOT/bin:$GOPATH/bin:$PATH
+          mage operator:testGo
+      - name: Login to GitHub Package Registry
+        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login docker.pkg.github.com -u "${{ github.actor }}" --password-stdin
+      - name: Login to ECR
         env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.ECR_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.ECR_SECRET }}
+        run: $(aws ecr get-login --no-include-email --region us-east-1)
+      - name: Setup Buildx
+        id: buildx
+        uses: crazy-max/ghaction-docker-buildx@v3
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        id: cache
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-
+      - name: Build and Push Docker
+        if: github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name == 'datastax/cass-operator'
+        env:
+          ECR_REPO: ${{ secrets.ECR_REPO }}
           PR_REF: ${{ github.event.pull_request.head.ref }}
-          MO_BASE_OS: 'registry.access.redhat.com/ubi7/ubi-minimal:7.8'
         run: |
           if [ "${GITHUB_EVENT_NAME}" == "pull_request" ]; then
             export MO_BRANCH=${PR_REF}
           else
             export MO_BRANCH="master"
-          fi;
-          export PATH=$GOROOT/bin:$GOPATH/bin:$PATH
-          mage operator:testAndBuild
-      - name: Deploy to ECR
-        if: github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name == 'datastax/cass-operator'
-        env:
-          MO_ECR_ID: ${{ secrets.ECR_ID }}
-          MO_ECR_SECRET: ${{ secrets.ECR_SECRET }}
-          MO_ECR_REPO: ${{ secrets.ECR_REPO }}
-        run: |
-          export PATH=$GOROOT/bin:$GOPATH/bin:$PATH
-          export MO_TAGS=$(cat ./build/tagsToPush.txt)
-          mage operator:deployToECR
-      - name: Deploy to GH Packages
-        if: github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name == 'datastax/cass-operator'
-        env:
-          MO_GH_USR: 'datastax/cass-operator'
-          MO_GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          MO_GH_PKG_REPO: 'datastax/cass-operator/operator'
-        run: |
+          fi
+
           export PATH=$GOROOT/bin:$GOPATH/bin:$PATH
-          export MO_TAGS=$(cat ./build/tagsToPush.txt)
-          mage operator:deployToGHPackages
+          GITHUB_REPO_URL="https://github.com/${{ github.repository }}"
```

Comment:
The following should look much like https://github.com/datastax/cass-config-builder/pull/19/files#diff-a1980bd700940451c303f902e86b140a in cass-config-builder as it does essentially the same thing.

Reply:
should we move long chunks of code here into a proper script in the repo?

Reply:
That's a good idea. I'll wait until I get the rest of the feedback and then move this into a script (or multiple scripts).